vimiv/utils/thumbnail_manager.py

```diff
@@ -72,14 +72,15 @@ def __init__(self, fail_pixmap: QPixmap, large: bool = True):
         xdg.makedirs(self.directory, self.fail_directory)
         self.fail_pixmap = fail_pixmap
 
-    def create_thumbnails_async(self, paths: List[str]) -> None:
+    def create_thumbnails_async(self, indices: Iterable[int], paths: List[str]) -> None:
```

Comment:
In the current implementation the indices are not always subsequent due to the threshold, right? Otherwise I would have said `paths: List[str], start: int = 0` and a later `enumerate` would be a lot easier, and make our life in the tests easier as well (passing the `[0]`, ... in all unit tests seems a bit clumsy.)

This does make me think if the `threshold` is really that useful, do you have it set above the interval size?

Reply:
Yes, I think I might have actually tried something like you described, before settling on this. And right, the setting should really only make a difference when set above the interval size.

It's intended for say... suddenly jumping to a position in the file list, then jumping back to the previous location, or something like that. Since there can be images left loaded out of range, they aren't suddenly loaded again in short order.

Granted, there would still be the delay of loading images from that sudden jump, in the first place, just not the jump back. And reads aren't a big deal on an SSD.

Reply:
The way indices are specified in unit tests is also cleaned up a little by 8d9118e358c0a9d757817e4d77ade834fa16c1d9

Reply:
Perfect, thank you!